charts/argo/templates/server-ingress.yaml

```diff
@@ -23,24 +25,37 @@ metadata:
     {{- end }}
 spec:
   rules:
-    {{- range .Values.server.ingress.hosts }}
-    - host: {{ . }}
+  {{- if .Values.server.ingress.hosts }}
+  {{- range $host := .Values.server.ingress.hosts }}
+    - host: {{ $host }}
       http:
         paths:
-          {{- if $.Values.server.ingress.paths }}
-          {{- range $.Values.server.ingress.paths }}
-          - backend:
-              serviceName: {{ .serviceName }}
-              servicePort: {{ .servicePort }}
-          {{- end }}
-          {{- end }}
-          - backend:
+  {{- if $extraPaths }}
+  {{- toYaml $extraPaths | nindent 10 }}
+  {{- end }}
+  {{- range $p := $paths }}
+          - path: {{ $p }}
+            backend:
               serviceName: {{ $serviceName }}
               servicePort: {{ $servicePort }}
-    {{- end -}}
+  {{- end -}}
+  {{- end -}}
+  {{- else }}
+    - http:
+        paths:
+  {{- if $extraPaths }}
+  {{- toYaml $extraPaths | nindent 10 }}
+  {{- end }}
+  {{- range $p := $paths }}
+          - path: {{ $p }}
+            backend:
+              serviceName: {{ $serviceName }}
```

Comment:
I believe `$serviceName` is correct as it is defined line 3, and the `$paths` value only contains the list of paths defined in the `values.yaml`.

The same strategy is used in the `argo-cd` charts: https://github.com/argoproj/argo-helm/blob/master/charts/argo-cd/templates/argocd-server/ingress.yaml#L51

Reply:
The `ingress.paths`element in the old `values.yaml` contains the list of (`serviceName`, `servicePort`) pairs. This seems to have been changed in your revision. The new code will break any clients that were using the old revision. I am not sure how useful (and used) the old version was, but breaking compatibility is not something to be taken lightly. If we end up taking this step (and I would like to get opinions from @stefansedich and @yann-soubeyrand on it), it should result in radical rather than incremental improvement in the chart quality. In this case, the old `paths` value was a misnomer, but having two very similar configuration options (`paths` and `extraPaths`) is confusing, and we should keep just one, IMHO.

Reply:
@vladlosev thanks for clarifying your point.

Yes I definitely agree that breaking changes should be scrutinized and having the thoughts of @stefansedich and @yann-soubeyrand would be very helpful on this.

My original aim was to add the possibility to be able to configure the ingress of this chart to serve `argo-workflow` on a path other than the root path, especially since the procedure is described in the `argo-workflow` docs (https://argoproj.github.io/argo-workflows/argo-server/#ingress).

I then chose to adopt the same logic than the `argo-cd` charts (https://github.com/argoproj/argo-helm/blob/master/charts/argo-cd/values.yaml#L504
), with the added bonus to harmonize all the Argo charts with ingresses, but as you say this indeed introduce a breaking change with the old values.yaml ...